libbeat/autodiscover/providers/docker/docker.go

```diff
@@ -192,12 +210,51 @@ func (d *Provider) generateMetaDocker(event bus.Event) (*docker.Container, *dock
 	return container, meta
 }
 
-func (d *Provider) emitContainer(event bus.Event, flag string) {
+func (d *Provider) startContainer(event bus.Event) {
+	container, meta := d.generateMetaDocker(event)
+	if container == nil || meta == nil {
+		return
+	}
+
+	if stopper, ok := d.stoppers[container.ID]; ok {
+		d.logger.Debugf("Container %s is restarting, aborting pending stop", container.ID)
+		stopper.Stop()
+		delete(d.stoppers, container.ID)
+		return
+	}
+
+	d.emitContainer(container, meta, "start")
+}
+
+func (d *Provider) scheduleStopContainer(event bus.Event) {
 	container, meta := d.generateMetaDocker(event)
 	if container == nil || meta == nil {
 		return
 	}
 
+	if d.config.CleanupTimeout <= 0 {
```

Comment:
Can user set a CleanupTimeout to be <= 0? Maybe this can be in some kind of a Validate() function for config?

Reply:
Added validate annotation for config, but I am going to keep the `<=` here because it feels safer :innocent: